src/query/sql/src/planner/binder/table.rs

```diff
@@ -352,63 +342,61 @@ impl Binder {
 
                 let op = StageTable::get_op(&user_stage_info)?;
 
-                let first_file = if files.is_empty() {
-                    let file = get_first_file(&op, &path).await?;
-                    match file {
-                        None => {
-                            return Err(ErrorCode::BadArguments(format!(
-                                "no file in {}",
-                                location
-                            )));
-                        }
-                        Some(f) => f.path().to_string(),
-                    }
-                } else {
-                    Path::new(&path)
-                        .join(&files[0])
-                        .to_string_lossy()
-                        .to_string()
-                };
-
-                let format_type = StageFileFormatType::Parquet;
-                let input_format = InputContext::get_input_format(&format_type)?;
-                let schema = input_format.infer_schema(&first_file, &op).await?;
-                user_stage_info.file_format_options = FileFormatOptions {
-                    format: format_type,
-                    record_delimiter: "".to_string(),
-                    field_delimiter: "".to_string(),
-                    nan_display: "".to_string(),
-                    skip_header: 0,
-                    escape: "".to_string(),
-                    compression: StageFileCompression::default(),
-                    row_tag: "".to_string(),
-                    quote: "".to_string(),
-                };
-                let stage_table_info = StageTableInfo {
-                    schema,
-                    user_stage_info,
-                    path: path.to_string(),
-                    files: vec![],
-                    pattern: Default::default(),
-                    files_to_copy: None,
-                };
+                if matches!(
+                    user_stage_info.file_format_options.format,
+                    StageFileFormatType::Parquet
+                ) {
+                    // Table functions always reside is default catalog
+                    let table_id = self
+                        .catalogs
+                        .get_catalog(CATALOG_DEFAULT)?
+                        .get_table_function_id("read_parquet")?;
+
+                    let files = list_file(&op, &path)
```

Comment:
I missed something here.
Please move the list_file to parquet table's `read_partitions` like what stage table did:
https://github.com/datafuselabs/databend/blob/54eb098effd0a737a844075203c3587eed755dc5/src/query/storages/stage/src/stage_table.rs#L137-L173

So we can make it distributed(cluster mode) easily, they are same the logic as fuse.
cc @youngsofun @RinChanNOWWW